zerver/lib/markdown/api_return_values_table_generator.py

```diff
@@ -59,11 +59,14 @@ def run(self, lines: List[str]) -> List[str]:
                 done = True
         return lines
 
-    def render_desc(self, description: str, spacing: int, return_value: Optional[str]=None) -> str:
+    def render_desc(self, description: str, spacing: int, data_type: str, return_value: Optional[str]=None) -> str:
         description = description.replace('\n', '\n' + ((spacing + 4) * ' '))
         if return_value is None:
-            return (spacing * " ") + "* " + description
-        return (spacing * " ") + "* `" + return_value + "`: " + description
+            arr = description.split(": ", 1)
+            if len(arr) == 1:
+                return (spacing * " ") + "* " + description
```

Comment:
What's the story with this split logic?  I don't understand why we'd need something like this.  

It sorta looks like you're trying to determine the API state by looking at whether the description starts with `foo:`, which 
I'm not sure is a pattern that makes sense to use.  Instead, we should be looking at some OpenAPI feature around the type, not text in the description.

Reply:
This logic is trying to extract the name of the ```additionalProperties``` from the description, as they don't have their variable names separately, but included in description, like 
```
 additionalProperties:
                          description: |
                            `{email_id}`: A list of the names of streams that
                            the user was subscribed to as a result of the query.
                          type: array
                          items:
                            type: string
```
For the other types of variables, the name is directly available in the ```return_value``` variable.

Reply:
Also, OpenAPI also renders it with description, so I'm not sure if there is some OpenAPI feature for it.
![image](https://user-images.githubusercontent.com/56172924/106851949-2f29b480-66dd-11eb-9562-b6ae486402e0.png)